### DISCUSSION TITLE
NCLR Support (5CLR, 6CLR, 7CLR, CMYK+N, etc.)

### DISCUSSION THREAD

**Discussion by ryanobeirne** [libvips/libvips#3045] [Help wanted]
I'm trying to convert sRGB images to a 6CLR (CMYK+Red+Green) device colorspace using 6CLR (0x36434c52) ICC profiles. Converting using `icc_export` to CMYK profiles works as expected, however, it appears that libvips does not support 6CLR profiles:

    $ vips icc_export input-rgb.tif output-6clr.tif --output-profile 6clr-profile.icc
    icc_export: unimplemented output color space 0x36434c52

Am I missing something, or is there another way to do this? I can provide images and profiles if it's helpful.

**Comment by jcupitt** [comment]
Hi @ryanobeirne,

I don't think I've ever tested more than CMYK output, so a sample profile would be great, thank you.

**Comment by ryanobeirne** [comment]
Thanks for the reply!

Here is a 6CLR icc profile, typical for what I work with:

[6clr-test.icc.zip](https://github.com/libvips/libvips/files/9550789/6clr-test.icc.zip)

**Comment by ryanobeirne** [comment]
I'm wondering if NCLR is even included in the TIFF spec. All the programs I use to generate 6CLR images for profiling targets kind of hack a CMYK TIFF with 2 additional spot channels:

![Screen Shot 2022-09-12 at 2 52 34 PM](https://user-images.githubusercontent.com/23461207/189733220-9ba7d0e0-dc8d-44be-be6b-2636c1de0fb4.png)

[6CLinCB-SP_P1.tiff.zip](https://github.com/libvips/libvips/files/9550960/6CLinCB-SP_P1.tiff.zip)

**Reply by jcupitt** [reply]
I made a super basic PR and it seems to work. Let's move discussion there.

Thanks for suggesting this improvement!

**Reply by ryanobeirne** [reply]
Thank you for this! I'll try it out soon.

This is one of the toughest issues I deal with as there isn't a lot of support for n-color in open source projects in general. And I haven't been able to find a starting point for troubleshooting. Thanks again!

#3046
